### ISSUE TITLE
Fix false positives on a block's last byte

### ISSUE THREAD

**Issue by lanjelot** [mwielgoszewski/python-paddingoracle#4]
Hey mate, thank you for publishing this neat Python port of PadBuster.

I've found a bug and submitted a PR to fix it.

When you ported PadBuster's Perl code to Python, you mis-translated this bit:
```    
if ($autoRetry == 1 &&  ($byteNum == ($blockSize - 1) ) && $hasHit == 0 ) {
  $hasHit++;
} else {
```

Sometimes the last byte you find is actually a false positive and as a result, you are not able to find the next byte. When this happens, you need to start from scratch for the entire block and skip over that false positive value you had for the last byte.

For example, use the below values to reproduce this issue (in DECRYPT mode):
```
key = '\x98\xc4Bt-c\xb4\t/\xb9S}\xbbj\xeb4'
iv = bytearray(b':j\x87\xf2x\xafU\r+M\xdaRh~\x91:')
ctext = 'tV\xe4\xda\xc2X\x13\x1e4\x1e\xf0\x1b\x1e\xfe\xc1\x1a\xb08\xebG\\#j\xbe\xd4w@7G\x0b\x06,\x1b\x1a\xbc8\xa6\xbd\xeb/]s\xe7\xc2\xe6_k\x86'
```
Which will produce the following error:
```
Traceback (most recent call last):
  File "paddingoracle-orig.py", line 359, in <module>
    test()
  File "paddingoracle-orig.py", line 304, in test
    decrypted = padbuster.decrypt(ctext, block_size=AES.block_size, iv=iv)
  File "paddingoracle-orig.py", line 135, in decrypt
    **kwargs)
  File "paddingoracle-orig.py", line 248, in bust
    byte_num, block, self.max_retries))
RuntimeError: Could not decrypt byte 13 in bytearray(b'\x1b\x1a\xbc8\xa6\xbd\xeb/]s\xe7\xc2\xe6_k\x86') within maximum allotted retries (3)
```

**Comment by mwielgoszewski** [comment]
Hey @lanjelot. Thanks for the bug report and PR! I'll provide any comments as I get through review.

**Comment by mwielgoszewski** [comment]
Just merged #5 - will push new wheel/egg to pypi in a day or two.

**Comment by mwielgoszewski** [comment]
Release 0.2.2 is now up on [PyPi](https://pypi.python.org/pypi/paddingoracle/0.2.2).
